### PR TITLE
Add mouse_move event support

### PR DESCRIPTION
**Pull request by znepb**
This completes the second to do task in [this](https://github.com/SquidDev-CC/CC-Tweaked/pull/434) pull request. For more information on this topic, please view that PR.

## Todo

- [ ]  Add TROR render support
- [X]  Add throttling
- [ ]  Make JFX work with throttling

### CODE REVIEW

**Comment by dmarcuse** [comment]
What happens when the window loses focus (e.g. with alt-tab)?

**Comment by znepb** [comment]
> What happens when the window loses focus (e.g. with alt-tab)?

When focus is lost, a `mouse_move` event is sent, with no parameters (nil nil). This is the same way it functions currently in Lemmmy's pull request.
As with alt-tab, it seems as right no events are sent.

File: plugin-api/src/main/java/net/clgd/ccemux/api/emulation/EmulatedComputer.java [hunk 1]
```diff
@@ -201,6 +201,17 @@ public void scroll(int lines, int x, int y) {
 		if (inTerminal(x, y)) queueEvent("mouse_scroll", new Object[] { lines, x, y });
 	}
 
+	/**
+	 * Queues a mouse move event
+	 */
+	public void move(int x, int y) {
+		if (inTerminal(x, y)) {
+			queueEvent("mouse_move", new Object[] { x, y });
```
Comment (Lemmmy) [inline plugin-api/src/main/java/net/clgd/ccemux/api/emulation/EmulatedComputer.java]: Where's the button arg?

**Review by Lemmmy — CHANGES_REQUESTED** [review]

**Comment by Lemmmy** [comment]
> This is the same way it functions currently in Lemmmy's pull request.

I don't think it does. This looks completely different. The mouse move throttle is in ticks, not milliseconds. It uses a monotonic clock and not System.currentTimeMillis. It passes a button arg to remain consistent with the other `mouse_` events. Did you even read the code at all?

File: plugin-api/src/main/java/net/clgd/ccemux/api/emulation/EmulatedComputer.java (hunk 1 above)
Comment (znepb) [reply plugin-api/src/main/java/net/clgd/ccemux/api/emulation/EmulatedComputer.java]: Mouse_move doesn’t need a button arg?

Comment (znepb) [reply plugin-api/src/main/java/net/clgd/ccemux/api/emulation/EmulatedComputer.java]: I added the button argument, it's always `1`.

**Comment by SquidDev** [comment]
See my comments in SquidDev-CC/CC-Tweaked#433.

**Pull request closed without merging**